### ISSUE TITLE
Remote Go Agent not automatically connecting to Go Server

### ISSUE THREAD

**Issue by countzero** [gocd/gocd#932]
We configured a `go-server` machine that runs a Go Server instance and a `go-agent-production` machine that runs a Go Agent. The Go Agent is configured to [auto register](http://www.go.cd/documentation/user/current/advanced_usage/agent_auto_register.html) itself to the Go Server. This works OK when the `go-server` machine comes up first  and then the `go-agent-production` machine starts.

The problem occurs when the `go-server` machine goes down (ie. has to reboot) and comes up again. Then the Go Agent `go-agent-production` gets listed under the "Agents" tab, but its status stays `pending` forever.

The only workaround in this situation is to also reboot the `go-agent-production` machine. It is __not__ sufficient to kill and restart all Go Agent related processes within the `go-agent-production` machine.

![fae47644-aa4d-11e4-851c-d998d88b1a7a](https://cloud.githubusercontent.com/assets/55295/6480156/8d773c34-c250-11e4-9721-56760e653b91.png)

**Comment by arvindsv** [comment]
You know, I've been thinking about this. This shouldn't happen. It should work properly. Did you find out anything new about it? Do you have the agent and bootstrapper logs from this?

**Comment by mohammed-amhend** [comment]
You should select the pending agent and click on enable button

**Comment by arvindsv** [comment]
Closing. No more information provided.

**Comment by sudhindra-glooko** [comment]
@arvindsv While autoregistering my agent retried and I was able to see the agent as pending on the agents view. When I tried to enable this agent from the new Agents view(from 16.10) it did not - basically suggested that Agent with [asdfkjasdkfjl] UUID not found. I switched back to the old UI and it let me enable the agent. Maybe a different issue but I googled and ran into this as a relevant ticket.

**Comment by arvindsv** [comment]
@sudhindra-glooko I think it's [this issue](https://github.com/gocd/gocd/issues/2762). It should be fixed in the next release.

**Comment by gregorskii** [comment]
Is it normal that one has to manually enable the agents?

**Comment by arvindsv** [comment]
@gregorskii Yes. Unless you want them to be [auto registered](https://docs.gocd.org/current/advanced_usage/agent_auto_register.html). If you're provisioning agents automatically, it makes sense to auto-register them.

**Comment by gregorskii** [comment]
I did set an auto register key. They connected to the server, but did not “enable” themselves on boot. I had to go into the agent screen and “enable” them. I have since found the api to enable them via curl.

**Comment by arvindsv** [comment]
Then it's possible the auto-register was not done properly. They will register and get enabled. We can take this over to a new issue, if you want, rather than here. However, I don't know of any issue with it.

**Comment by gregorskii** [comment]
Ok, I’ll spend some more time looking at it in the next few days, and open an issue if I can’t resolve.

Appreciate it.
